rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -444,7 +444,20 @@ void main_window::InstallPup()
 		updatefilenames.begin(), updatefilenames.end(), [](std::string s) { return s.find("dev_flash_") == std::string::npos; }),
 		updatefilenames.end());
 
-	QProgressDialog pdlg(tr("Installing firmware ... please wait ..."), tr("Cancel"), 0, static_cast<int>(updatefilenames.size()), this);
+	std::string version_string = pup.get_file(0x100).to_string();
+	version_string.erase(version_string.find('\n'));
+
+	const float cur_version = 4.81f;
+	float version = std::stof(version_string);
```

Comment:
Function stof is locale dependant.
The value of the variable _version_ is equal to 4 on my system as it doesn't use decimal-point character (the variable _version_string_ contains "4.81").
The comparison below fails and I have the message box with the text "Old firmware detected [...]" whereas I have used the latest firmware available.